### DISCUSSION TITLE
Pagination help

### DISCUSSION THREAD

**Discussion by nomanmurtaza786** [supabase/supabase#2139] [General]
Hello Guy, how I can show no more data thing with rang function?

**Comment by silentworks** [comment]
Take a look at my answer on this topic in this other discussion.

https://github.com/supabase/supabase/discussions/1223#discussioncomment-641447

**Reply by nomanmurtaza786** [reply]
Thanks mate, i have seen your answer, same things are available in flutter/ dart? 

I have suggested can do make a trigger update total count on insert and delete.  We can refer to that value instead of count. Can you help to write that trigger. 
I think Count function is heavy for performance.

**Reply by silentworks** [reply]
That feels like premature optimisation, I don't think that is something you need to worry about at this stage. Also making two restful calls might be a longer roundtrip than a count function taking long due to the number of records.

**Reply by steve-chavez** [reply]
> I think Count function is heavy for performance.

@nomanmurtaza786 @silentworks There's also the `count: "planned"` [option](https://supabase.io/docs/reference/javascript/select#querying-with-count-option), which uses the PostgreSQL planner statistics for the count - this one will be fast even with a big amount of rows.
